<launch>
  <!-- 全局参数配置 -->
  <arg name="num_agents" default="2" />
  <arg name="debug_mode" default="False"/>
  <rosparam param="dx">[0.0, 0]</rosparam> 
  <rosparam param="dy">[0.0, 30]</rosparam>

  <!-- 机器人1配置 -->
  <arg name="robot3_rgb_topic" default="/robot3/spark_camera/image_raw/compressed" />
  <arg name="robot3_depth_topic" default='/robot3/spark_camera/depth/image_raw/compressedDepth' />
  <arg name="robot3_tf_source" default="robot3/base_link" />
  <arg name="robot3_tf_target" default="robot3/map" />
  <arg name="robot3_window_size_x" default="20" />
  <arg name="robot3_window_size_y" default="20" />
  <arg name="robot3_window_size_yaw" default="20" />
  
  <!-- 机器人1配置 -->
  <arg name="robot1_rgb_topic" default="/robot1/camera/color/image_raw/compressed" />
  <arg name="robot1_depth_topic" default='/robot1/camera/aligned_depth_to_color/image_raw/compressedDepth' />
  <arg name="robot1_tf_source" default="robot1/base_link" />
  <arg name="robot1_tf_target" default="robot1/map" />
  <arg name="robot1_window_size_x" default="20" />
  <arg name="robot1_window_size_y" default="20" />
  <arg name="robot1_window_size_yaw" default="20" />
  
  
  <!-- 机器人1节点 -->
  <node name="robot1_collector" pkg="explore2" type="message_collector_syn.py"  output="screen" >
    <param name="robot_id" value="0" />
    <param name="topic_config/rgb" value="$(arg robot3_rgb_topic)" />
    <param name="topic_config/depth" value="$(arg robot3_depth_topic)" />
    <param name="topic_config/rgb_type" value="CompressedImage" />
    <param name="topic_config/depth_type" value="CompressedImage" />
    <param name="tf_config/target_frame" value="$(arg robot3_tf_target)" />
    <param name="tf_config/source_frame" value="$(arg robot3_tf_source)" />
    <param name="filter_config/window_size_x" value="$(arg robot3_window_size_x)" />
    <param name="filter_config/window_size_y" value="$(arg robot3_window_size_y)" />
    <param name="filter_config/window_size_yaw" value="$(arg robot3_window_size_yaw)" />
    <param name="filter_config/display_filter" value="false" />
  </node>
  
  <!-- 机器人2节点 -->
  <node name="robot2_collector" pkg="explore2" type="message_collector_syn.py"  output="screen">
    <param name="robot_id" value="1" />
    <param name="topic_config/rgb" value="$(arg robot3_rgb_topic)" />
    <param name="topic_config/depth" value="$(arg robot3_depth_topic)" />
    <param name="topic_config/rgb_type" value="CompressedImage" />
    <param name="topic_config/depth_type" value="CompressedImage" />
    <param name="tf_config/target_frame" value="$(arg robot3_tf_target)" />
    <param name="tf_config/source_frame" value="$(arg robot3_tf_source)" />
    <param name="filter_config/window_size_x" value="$(arg robot3_window_size_x)" />
    <param name="filter_config/window_size_y" value="$(arg robot3_window_size_y)" />
    <param name="filter_config/window_size_yaw" value="$(arg robot3_window_size_yaw)" />
    <param name="filter_config/display_filter" value="false" />
  </node>
  
  <!-- 主节点 结点名称会影响config的效果-->
  <node name="mr_planning_node" pkg="explore2" type="mr_planning_1r.py" output="screen">
    <param name="num_agents" value="$(arg num_agents)" />
    <param name="debug" value="$(arg debug_mode)" />
  </node>

  <!-- <node name="rviz_spark" pkg="rviz" type="rviz" args="-d $(find explore2)/rviz_config/spark.rviz" required="false" /> -->
  <!-- <node name="rviz_mr600" pkg="rviz" type="rviz" args="-d $(find explore2)/rviz_config/mr600.rviz" required="false" />
  
  <node name="rviz_petbot" pkg="rviz" type="rviz" args="-d $(find explore2)/rviz_config/petbot.rviz" required="false" /> -->
</launch>
